js/src/evaluation/llm_evaluator.ts

```diff
@@ -0,0 +1,290 @@
+  reasoningKey?: string;
+  /** Optional description of score reasoning, provided to the LLM in the structured output */
+  reasoningDescription?: string;
+
+  /**
+   * Creates a new continuous score configuration
+   * @param params Configuration parameters
+   * @param params.key Feedback key for the evaluator
+   * @param params.description Description of the scoring criteria
+   * @param params.min Optional minimum score value (defaults to 0)
+   * @param params.max Optional maximum score value (defaults to 1)
+   * @param params.reasoningKey Optional key for the LLM reasoning/explanation for the score
+   * @param params.reasoningDescription Optional description of score reasoning, provided to the LLM in the structured output
+   */
+  constructor(params: {
+    key: string;
+    description: string;
+    min?: number;
+    max?: number;
+    reasoningKey?: string;
+    reasoningDescription?: string;
+  }) {
+    this.key = params.key;
+    this.min = params.min ?? 0;
+    this.max = params.max ?? 1;
+    this.description = params.description;
+    this.reasoningKey = params.reasoningKey;
+    this.reasoningDescription = params.reasoningDescription;
+  }
+}
+
+type ScoreConfig = CategoricalScoreConfig | ContinuousScoreConfig;
+
+function createScoreJsonSchema(scoreConfig: ScoreConfig): Record<string, any> {
+  const properties: Record<string, any> = {};
+
+  if (scoreConfig.reasoningKey) {
+    properties[scoreConfig.reasoningKey] = {
+      type: "string",
+      description:
+        scoreConfig.reasoningDescription ||
+        "First, think step by step to explain your score.",
+    };
+  }
+
+  if ("choices" in scoreConfig) {
+    properties.value = {
+      type: "string",
+      enum: scoreConfig.choices,
+      description: `The score for the evaluation, one of ${scoreConfig.choices.join(
+        ", "
+      )}.`,
+    };
+  } else {
+    properties.score = {
+      type: "number",
+      minimum: scoreConfig.min,
+      maximum: scoreConfig.max,
+      description: `The score for the evaluation, between ${scoreConfig.min} and ${scoreConfig.max}, inclusive.`,
+    };
+  }
+
+  return {
+    title: scoreConfig.key,
+    description: scoreConfig.description,
+    type: "object",
+    properties,
+    required: scoreConfig.reasoningKey
+      ? ["choices" in scoreConfig ? "value" : "score", scoreConfig.reasoningKey]
+      : ["choices" in scoreConfig ? "value" : "score"],
+  };
+}
+
+interface LLMEvaluatorParams {
+  promptTemplate: string | [string, string][];
+  scoreConfig: ScoreConfig;
+  chatModel: BaseLanguageModel;
+  mapVariables?: (run: Run, example?: Example) => Record<string, any>;
+}
+
+export class LLMEvaluator implements RunEvaluator {
+  prompt: any;
+  mapVariables?: (run: Run, example?: Example) => Record<string, any>;
+  scoreConfig: ScoreConfig;
+  scoreSchema: Record<string, any>;
+  runnable: any;
+
+  constructor() {}
+
+  static async create(params: LLMEvaluatorParams): Promise<LLMEvaluator> {
```

Comment:
You definitely shouldn't have two methods for this, ideally you just use the constructor